app/scripts/services/dimItemService.factory.js

```diff
@@ -289,64 +289,58 @@
       function canEquipExotic(item, store) {
         var deferred = $q.defer();
         var promise = deferred.promise;
-        var hasLifeExotic = _.contains(item.talentPerks, 4044819214);
 
-        var prefix = _(store.items)
-          .chain()
-          .filter(function(i) {
-            return (i.equipped && i.type !== item.type && i.sort === item.sort && i.tier === dimItemTier.exotic)
+        var equippedExotics = _.filter(store.items, function(i) {
+            return (i.equipped &&
+                    i.type !== item.type &&
+                    i.sort === item.sort &&
+                    i.tier === 'Exotic');
           });
 
-        var amount = prefix.size().value();
-
         // Fix for "The Life Exotic" Perk on Exotic Items
         // Can equip multiples
+        function hasLifeExotic(item) {
+          return _.contains(item.talentPerks, 4044819214);
+        }
 
-        if (amount === 0) {
+        if (equippedExotics.length === 0) {
           deferred.resolve(true);
-        } else if (amount === 1) {
-          var exoticItem = prefix.value()[0];
+        } else if (equippedExotics.length === 1) {
+          var equippedExotic = equippedExotics[0];
 
-          if (hasLifeExotic) {
-            deferred.resolve(true);
-          } else if (_.contains(exoticItem.talentPerks, 4044819214)) {
+          if (hasLifeExotic(item) || hasLifeExotic(equippedExotic)) {
```

Comment:
I did a bunch of cleanup to the code, but this was the critical change. Before, we only allowed double-exotics when the item you were trying to equip had the Life Exotic perk. Now, we allow it if the item you're trying to equip *or the already equipped item* has the Life Exotic perk.